src/plexil-adapter/OwInterface.cpp

```diff
@@ -746,140 +646,235 @@ void OwInterface::takePicture (int id)
 {
   if (! mark_operation_running (Op_TakePicture, id)) return;
   std_msgs::Empty msg;
-  ROS_INFO ("  Capturing stereo image using left image trigger.");
+  ROS_INFO ("Capturing stereo image using left image trigger.");
   thread fault_thread (monitor_for_faults, Op_TakePicture);
   fault_thread.detach();
   m_leftImageTriggerPublisher->publish (msg);
 }
 
+void OwInterface::deliver (double x, double y, double z, int id)
+{
+  if (! mark_operation_running (Op_Deliver, id)) return;
+  thread action_thread (&OwInterface::deliverAction, this, x, y, z, id);
+  action_thread.detach();
+}
+
+template <int OpIndex, class ActionClient, class Goal,
+          class ResultPtr, class FeedbackPtr>
+void OwInterface::runAction (const string& opname,
+                             std::unique_ptr<ActionClient>& ac,
+                             const Goal& goal, int id)
+{
+  thread fault_thread (monitor_for_faults, opname);
+  if (ac) {
+    ac->sendGoal (goal,
+                  action_done_cb<OpIndex, ResultPtr>,
+                  active_cb<OpIndex>,
+                  action_feedback_cb<FeedbackPtr>);
```

Comment:
Related to a note made earlier about the un-necessity of making the second arg of `action_done_cb` as a typename and the same for `action_feedback_cb`
```suggestion
                  action_done_cb<OpIndex>,
                  active_cb<OpIndex>,
                  action_feedback_cb);
```

Reply:
See comments above about the removal of templates not working.